### PR TITLE
Ruff linting

### PR DESCRIPTION
**Pull request by ljgray**
Adding ruff linting here. See https://github.com/radiocosmology/draco/pull/243

### CODE REVIEW

File: ch_pipeline/analysis/decorrelation.py [hunk 1]
```diff
@@ -21,14 +21,12 @@
   of the timestream data
 """
 # === Start Python 2/3 compatibility
-from __future__ import absolute_import, division, print_function, unicode_literals
 from future.builtins import *  # noqa  pylint: disable=W0401, W0614
 from future.builtins.disabled import *  # noqa  pylint: disable=W0401, W0614
 
 # === End Python 2/3 compatibility
```
Comment (ketiltrout) [inline ch_pipeline/analysis/decorrelation.py:27]: Presumably this whole thing should go, since `python_requires` is `>=3.9`.

Comment (ljgray) [reply ch_pipeline/analysis/decorrelation.py:27]: Good catch

File: ch_pipeline/hfb/containers.py [hunk 2]
```diff
@@ -323,7 +316,9 @@ class HFBHighResContainer(FreqContainer, HFBContainer):
 class HFBHighResData(TODContainer, HFBHighResContainer, HFBBeamContainer):
     """Container for holding high-resolution frequency data."""
 
-    _dataset_spec = {
+    _axes = ("beam",)
```
Comment (rikvl) [inline ch_pipeline/hfb/containers.py:319]: The `beam` axis is already added via `HFBBeamContainer`

File: ch_pipeline/hfb/containers.py [hunk 3]
```diff
@@ -350,7 +345,9 @@ class HFBHighResData(TODContainer, HFBHighResContainer, HFBBeamContainer):
 class HFBHighResTimeAverage(HFBHighResContainer, HFBBeamContainer):
     """Container for holding time-averaged high-resolution frequency data."""
 
-    _dataset_spec = {
+    _axes = ("beam",)
```
Comment (rikvl) [inline ch_pipeline/hfb/containers.py:348]: Idem. (The `beam` axis is already added via `HFBBeamContainer`)

File: ch_pipeline/hfb/io.py [hunk 4]
```diff
@@ -252,9 +241,7 @@ def _find_beam(self):
         beams_xy = mdl.get_beam_positions(beams_ind, freq).squeeze()
 
         # Find NS beam number of beam closest to calibration source
-        beam_index_ns = np.abs(beams_xy[:, 1] - src_y).argmin()
-
-        return beam_index_ns
+        return np.abs(beams_xy[:, 1] - src_y).argmin()
```
Comment (rikvl) [inline ch_pipeline/hfb/io.py:244]: It seems to me something useful gets lost when not allowing the result of a calculation to be stored in a variable before returning it:
- The name of the variable conveys some information.
- The `Returns` section in the docstring refers to this variable.

I guess this is an opinion of the ruff linter that I'll just have to get used to (unnecessary-assign (RET504))

**Comment by rikvl** [comment]
I looked through the changes in the HFB code in detail. That's all good except for adding the `beam` axis in two containers that should already inherit it from `HFBBeamContainer`. Or maybe I misunderstand something about how containers work.

I still have to go through the rest of the PR

**Comment by ljgray** [comment]
> I looked through the changes in the HFB code in detail. That's all good except for adding the `beam` axis in two containers that should already inherit it from `HFBBeamContainer`. Or maybe I misunderstand something about how containers work.
> 
> I still have to go through the rest of the PR

I'm not sure where that came from - this branch has been open for awhile so it may have been based of off some other branch at some point. I'll just remove those because you're correct, they shouldn't need a beam axis

**Comment by ketiltrout** [comment]
Tests are failing because you need to add the `Install apt dependencies` task to the ruff check workflow.  (The old black check didn't install the package).

**Comment by ketiltrout** [comment]
This might be a downside of using `optional-dependencies` to install the linters: it forces an otherwise unnecessary install of the package itself.  Might be better just to manually install ruff and black in the workflow.

File: pyproject.toml [hunk 5]
```diff
@@ -49,3 +50,49 @@ readme = {file = ["README.md"], content-type = "text/markdown"}
...
+# Ignore the following directories
+exclude = [
+    ".git",
+    ".github",
+    "build",
+    "doc",
+    "test",
+    "setup.py",
+    "versioneer.py",
+    "*/_version.py",
```
Comment (ketiltrout) [inline pyproject.toml:94]:
```suggestion
```
No longer present.

File: ch_pipeline/analysis/sidereal.py [hunk 6]
```diff
@@ -524,14 +519,16 @@ def get_times(acq_files):
...
         # Load in file (but ignore all datasets)
         ad_empty = andata.AnData.from_acq_h5(acq_files, datasets=())
         start = ad_empty.timestamp[0]
         end = ad_empty.timestamp[-1]
+
         return start, end
-    else:
-        raise Exception("Input %s, not understood" % repr(acq_files))
+
+    raise Exception(f"Input {acq_files!r}, not understood")
```
Comment (ketiltrout) [inline ch_pipeline/analysis/sidereal.py:531]:
```suggestion
    raise TypeError(f"Input {acq_files!r}, not understood")
```

File: ch_pipeline/hfb/analysis.py [hunk 7]
```diff
@@ -1368,7 +1340,7 @@ def _interpolation_linear(x, y, w, xeval, mode="zero", xperiod=None):
 
 def _ensure_list(x):
     if hasattr(x, "__iter__"):
-        y = [xx for xx in x]
+        y = [xx for xx in x]  # noqa: C416
```
Comment (ketiltrout) [inline ch_pipeline/hfb/analysis.py:1343]:
```suggestion
        y = list(x)
```

File: ch_pipeline/processing/base.py [hunk 8]
```diff
@@ -655,14 +652,14 @@ def classify_failed(
         # See if any of the patterns that we are looking for
         # exist in the stdout
         for key, regex_patterns in patterns.items():
-            if any([bool(re.search(p, tail)) for p in regex_patterns]):
+            if any([bool(re.search(p, tail)) for p in regex_patterns]):  # noqa: C419
```
Comment (ketiltrout) [inline ch_pipeline/processing/base.py:655]:
```suggestion
            if any(bool(re.search(p, tail)) for p in regex_patterns):
```

**Review by ketiltrout — APPROVED** [review]
I have a few optional niggles for you:

**Pull request merged**